src/commands.ts

```diff
@@ -66,53 +79,50 @@ export class Commands {
 	 */
 	public async login(args?: {
 		url?: string;
-		token?: string;
-		label?: string;
+		safeHostname?: string;
 		autoLogin?: boolean;
 	}): Promise<void> {
-		if (this.contextManager.get("coder.authenticated")) {
+		if (this.deploymentManager.isAuthenticated()) {
 			return;
 		}
 		this.logger.info("Logging in");
 
-		const url = await maybeAskUrl(this.mementoManager, args?.url);
+		const currentDeployment = await this.secretsManager.getCurrentDeployment();
+		const url = await maybeAskUrl(
+			this.mementoManager,
+			args?.url,
+			currentDeployment?.url,
+		);
 		if (!url) {
-			return; // The user aborted.
+			return;
 		}
 
 		// It is possible that we are trying to log into an old-style host, in which
-		// case we want to write with the provided blank label instead of generating
-		// a host label.
-		const label = args?.label === undefined ? toSafeHost(url) : args.label;
-
-		// Try to get a token from the user, if we need one, and their user.
-		const autoLogin = args?.autoLogin === true;
-		const res = await this.maybeAskToken(url, args?.token, autoLogin);
-		if (!res) {
-			return; // The user aborted, or unable to auth.
-		}
-
-		// The URL is good and the token is either good or not required; authorize
-		// the global client.
-		this.restClient.setHost(url);
-		this.restClient.setSessionToken(res.token);
-
-		// Store these to be used in later sessions.
-		await this.mementoManager.setUrl(url);
-		await this.secretsManager.setSessionToken(res.token);
-
-		// Store on disk to be used by the cli.
-		await this.cliManager.configure(label, url, res.token);
+		// case we want to write with the provided blank hostname instead of
+		// generating one.
+		const safeHostname = args?.safeHostname ?? toSafeHost(url);
+		this.logger.info("Using hostname", safeHostname);
+
+		const result = await this.loginCoordinator.promptForLogin({
```

Comment:
The function name implies to me this will always be a prompt, but it is conditional.  I was using the `maybe` prefix to indicate this but there are any number of names we could use I think.